include/envoy/router/router.h

```diff
@@ -142,6 +187,11 @@ class VirtualHost {
   virtual ~VirtualHost() {}
 
   /**
+   * @return CorsPolicySharedPtr the CORS policy for this virtual host.
+   */
+  virtual const CorsPolicySharedPtr corsPolicy() const PURE;
```

Comment:
Any particular reason to return a shared_ptr here? I think we can likely just return a `const CorsPolicy*` to avoid the shared_ptr inc/dec. Same below. If this is the case we can remove CorsPolicySharedPtr typedef.

Reply:
My first version had `CorsPolicy*`, but we weren't sure if that survives config reloads etc so I was advised to change it to shared pointers. Maybe I am doing it wrong. I thought to use shared pointers you need to create them when you do the `new` so it's all the way to the config.
Should I change it back?

Reply:
It's safe to use a raw pointer here as the route table that backs it is snapped for the lifetime of the request. So I would recommend switching it back.

Reply:
Ok, maybe I can just remove that commit :)

Reply:
After a second thought, rebasing deletes all history here so I probably better don't just remove the commit.